### DISCUSSION TITLE
`npx cap run android` doesn't seem to do anything

### DISCUSSION THREAD

**Discussion by Harvtronix** [ionic-team/capacitor#4230] [Q&A]
Hello! I'm super excited to be using the capacitor and ionic frameworks. I got my first capacitor app running on both android and ios in under an hour, which was awesome!

I'm having a bit of an issue though, so I'm curious if I'm doing something incorrectly or if perhaps I've hit a legitimate bug.

I'm using Capacitor 3.0.0-beta.3.

When I run `npx cap run android`, I get the following output:

```
(geodude)% npx cap run android
✔ Copying web assets from build to android/app/src/main/assets/public in 11.65ms
✔ Creating capacitor.config.json in android/app/src/main/assets in 1.56ms
✔ copy android in 29.75ms
✔ Updating Android plugins in 3.91ms
✔ update android in 39.89ms
(geodude)% 
```

But then the terminal prompt just returns and nothing else ever happens. When I do the same for ios, it prompts me to select a virtual device and then launches the app in it. I'm not seeing the same sort of thing for android.

`npx cap open android` works as expected and I can click the green "run" button in there to launch the app in a virtual device with no issues.

For fun, I saw there was a `--target` command line option to `cap run`, so I tried passing my virtual device's id to it:

```
(geodude)% npx cap run --target Pixel_3a_API_30 android
✔ Copying web assets from build to android/app/src/main/assets/public in 14.54ms
✔ Creating capacitor.config.json in android/app/src/main/assets in 1.78ms
✔ copy android in 32.80ms
✔ Updating Android plugins in 4.46ms
✔ update android in 43.30ms
[error] Invalid target ID: Pixel_3a_API_30.
        Valid targets are:
(geodude)% 
```

I also just tried listing the virtual devices from the command line via `advmanager` (not sure if this is now `cap` is doing it), and my device does indeed show up in the list.

Any help would be appreciated!

I didn't see any debug-type arguments for `cap`, but if they are available, I'm more than happy to turn them on for additional info.

**Comment by alanxp** [comment]
i get gradle errors aswell

**Reply by thomasvidas** [reply]
For Capacitor 3, are you on Gradle 7 or higher? It is documented in the upgrade guide that gradle 7 is the new minimum

**Reply by dsuttonpreece** [reply]
I upgraded my Capacitor 3 project to use Gradle 7 thinking that might help, but now I'm getting a Gradle error that I wasn't getting with 6.5…  

Running cap run still exits before showing any emulator targets:
```
❯ npx cap run android
✔ Copying web assets from build to android/app/src/main/assets/public in 59.21ms
✔ Creating capacitor.config.json in android/app/src/main/assets in 1.22ms
[info] Found 1 Cordova plugin for android:
       cordova-plugin-keychain-storage@1.1.5
✔ copy android in 98.71ms
✔ Updating Android plugins in 4.44ms
[info] Found 2 Capacitor plugins for android:
       @capacitor/app@0.3.8
       @capacitor/browser@0.4.7
[info] Found 1 Cordova plugin for android:
       cordova-plugin-keychain-storage@1.1.5
✔ update android in 137.57ms
```
But when I try build the project `cap run` has introduced an error:
```
Could not compile build file '<project_root>/android/capacitor-cordova-android-plugins/build.gradle'.
> startup failed:
  build file '<project_root>/android/capacitor-cordova-android-plugins/build.gradle': 21: unable to resolve class targetSdkVersion 
   @ line 21, column 26.
             targetSdkVersion targetSdkVersion = project.hasProperty('targetSdkVersion') ? rootProject.ext.targetSdkVersion : 30
```
Manually fixing the error means I can build the project again. But running `cap run` reintroduces the error every time.

**Reply by thelement115** [reply]
did you find a solution?, I am getting the same error

**Reply by alanxp** [reply]
> did you find a solution?, I am getting the same error

Downgraded to v2. So many problems with 3.0

**Reply by Feffy** [reply]
I had the same issue. Installing the latest capacitor cli version and then running `npx cap sync` fixed it. Hope this helps.

**Reply by Danbardo** [reply]
I had a similar problem, I just left mine for about 5 minutes and it asked to select a device.

**Reply by c-y-q** [reply]
change java jdk version, clean as and restore as, l have tired it ,is ok

**Comment by dsuttonpreece** [comment]
@Harvtronix I'm also having this issue. Device list is empty when `npx cap run --list android`, but `avdmanager list` returns a long list of devices…

Do you intend on raising this as an issue?

**Comment by brunocrosier** [comment]
Same issue, running `npx cap run --list android` doesn't seem to do anything. I think the [docs](https://capacitorjs.com/docs/cli/run) might benefit from a bit more detail here

**Comment by Harvtronix** [comment]
Looks like someone opened issue #4571 for this. Looks related to Android SDK 30 vs 29. I'll monitor that and hope for a solution soon!

**Comment by jcesarmobile** [comment]
this was a bug in native-run, has been fixed in latest version (1.4.0)

since native-run is a dependency of @capacitor/cli, you can try to reinstall the CLI and it should pick the latest version, but make sure that was the case by checking the installed version in package-lock.json
If not updated, you can force update the dependency with npm update native-run --depth 6

**Reply by ravithb** [reply]
Can confirm that it still happens even with native-run 1.4.0

**Reply by jcesarmobile** [reply]
Update capacitor CLI to 3.0.1 and try again.
It included some improvements to show native-run errors.
This was fixed in 1.4.0, so should be something else.
